src/main/java/com/sessions/session9/homeworkOOP2/User.java

```diff
@@ -0,0 +1,158 @@
+package com.sessions.session9.homeworkOOP2;
+
+import java.time.LocalDate;
+import java.util.Objects;
+import java.util.Random;
+
+public class User implements Comparable<User>
+{
+    private int age;
+    private String name;
+    private LocalDate dateOfBirth;
+    private Nationality nationality;
+    private boolean hasJob;
+    private final int idUser;
+    private static int nrOfUsers = 0;
+    private static int incrementIDs = 1;
+
+    private int autoGeneratedAge(int origin, int bound)
+    {
+        Random random = new Random();
+
+        return random.nextInt(origin, bound);
+    }
+
+    public User()
+    {
+        age = autoGeneratedAge(18, 80);
+        name = "Unkown";
+        dateOfBirth = LocalDate.now();
+        nationality = Nationality.Unknown;
+        hasJob = false;
+        idUser = incrementIDs++;
+        nrOfUsers++;
+    }
+
+    public User(final int age, final String name, final LocalDate dateOfBirth, final Nationality nationality, final boolean hasJob)
+    {
+        this.age = age;
+        this.name = name;
+        this.dateOfBirth = dateOfBirth;
+        this.nationality = nationality;
+        this.hasJob = hasJob;
+        this.idUser = incrementIDs++;
```

Comment:
Recomand sa te uiti putin peste UUID-uri, in practica se folosesc ca id-uri ale entitatilor. Au un format ceva mai mare, ceea ce inseamna ca e greu pentru cineva sa "ghiceasca" un it si totodata se asigura sanse mici de coliziune (doua id-uri sa fie la fel), asta e important pentru ca id-urile ajung sa fie folosite ca si chei primare in db. Chiar daca in cazul tau nu ai o problema pentru ca mereu incrementezi id-ul respectiv 1,2,3... m-as uita si peste documentatia de mai jos care descrie folosirea de UUIDuri in Java:
https://www.baeldung.com/java-uuid